src/main/host/process.rs

```diff
@@ -1551,4 +1557,143 @@ mod export {
         })
         .unwrap();
     }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn _process_threadLeader(
+        proc: *const RustProcess,
+    ) -> *mut cshadow::Thread {
+        let proc = unsafe { proc.as_ref().unwrap() };
+        Worker::with_active_host(|host| {
+            let proc = proc.borrow(host.root());
+            let tid = ThreadId::from(proc.id());
+            let threads = proc.threads.borrow();
+            match threads.get(&tid) {
+                Some(t) => unsafe { t.cthread() },
+                None => std::ptr::null_mut(),
+            }
+        })
+        .unwrap()
+    }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn _process_getThread(
+        proc: *const RustProcess,
+        tid: libc::pid_t,
+    ) -> *mut cshadow::Thread {
+        let proc = unsafe { proc.as_ref().unwrap() };
+        Worker::with_active_host(|host| {
+            let proc = proc.borrow(host.root());
+            let tid = ThreadId::try_from(tid).unwrap();
+            let threads = proc.threads.borrow();
+            match threads.get(&tid) {
+                Some(t) => unsafe { t.cthread() },
+                None => std::ptr::null_mut(),
+            }
+        })
+        .unwrap()
+    }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn _process_removeThread(proc: *const RustProcess, tid: libc::pid_t) {
+        let proc = unsafe { proc.as_ref().unwrap() };
+        Worker::with_active_host(|host| {
+            let proc = proc.borrow(host.root());
+            let tid = ThreadId::try_from(tid).unwrap();
+            proc.threads.borrow_mut().remove(&tid);
+        })
+        .unwrap()
+    }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn _process_insertThread(
+        proc: *const RustProcess,
+        thread: *mut cshadow::Thread,
+    ) {
+        let proc = unsafe { proc.as_ref().unwrap() };
+        Worker::with_active_host(|host| {
+            let proc = proc.borrow(host.root());
+            let thread = unsafe { ThreadRef::new(thread) };
+            let tid = thread.id();
+            proc.threads.borrow_mut().insert(tid, thread);
+        })
+        .unwrap()
+    }
```

Comment:
Could use a comment explaining if it takes ownership of the thread.

Reply:
Done